nntrainer/models/neuralnet.h

```diff
@@ -354,6 +357,70 @@ class NeuralNetwork : public ml::train::Model {
     manager.setGradientMemoryOptimization(opt);
   }
 
+/// @todo Make a more common class have this
+/// Maybe appcontext can have this?
+#ifdef PROFILE
+  class Profiler {
+  public:
+    // Designated key.
+    enum { SINGLE_FORWARD = 0 };
```

Comment:
is this SINGLE_FORWARD intended to give time for a single forward. If yes, then maybe divide by count maybe needed. If not, then maybe the SINGLE prefix can be removed?

Reply:
I think it is a bit misleading, changed.